### PR TITLE
add a new example: video-annotation

### PR DESCRIPTION
**Pull request by xx205**
This PR gives a new example: video annotation, which is useful in some cases. For example, partitioning an audio stream into homogeneous segments according to the speaker identity is difficult, especially when the number of speakers is large, but it will be much easier with visual cues.

### CODE REVIEW

File: example/media/nasa.mp4 [hunk 1]
```diff
@@ -0,0 +1 @@
+Happy Thanksgiving From the International Space Station.mp4
```
Comment (thijstriemstra) [inline example/media/nasa.mp4]: this isn't a mp4 file..

File: example/video-annotation/index.html [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/styles/default.min.css">
+        <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/highlight.min.js"></script>
+        <script>hljs.initHighlightingOnLoad();</script>
+    </head>
+
+    <body itemscope itemtype="http://schema.org/WebApplication">
+        <div class="container">
+            <!-- <div class="header">
+                <h1 itemprop="name">Video Annotation System</h1>
+            </div> -->
```
Comment (thijstriemstra) [inline example/video-annotation/index.html]: bring back the header or remove the dead code..

File: example/video-annotation/index.html [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    
+                    <div class="col-sm-3">
+                        <button class="btn btn-info btn-block" data-action="export" title="Export annotations to JSON">
+                            <i class="glyphicon glyphicon-file"></i>
+                            Export
+                        </button>
+                    </div>
+                </div>
+
+                <!-- <div class="controls">
```
Comment (thijstriemstra) [inline example/video-annotation/index.html]: bring back the controls or remove the dead code..

File: example/video-annotation/main.js [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        normalize: true,
+        splitChannels: false,
+        backend: 'MediaElement',
+        plugins: [
+            WaveSurfer.regions.create(),
+            WaveSurfer.minimap.create({
+                height: 30,
+                waveColor: '#ddd',
+                progressColor: '#999'
+                // cursorColor: '#999'
```
Comment (thijstriemstra) [inline example/video-annotation/main.js]: remove the dead code.

File: example/video-annotation/main.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+            .ajax({
+                responseType: 'json',
+                url: '../media/nasa.json'
+            })
+            .on('success', function(data) {
+                loadRegions(data);
+                saveRegions();
+            });
+    });
+    // wavesurfer.on('region-click', function(region, e) {
```
Comment (thijstriemstra) [inline example/video-annotation/main.js]: remove the dead code.

File: example/video-annotation/main.js [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+    wavesurfer.on('region-out', hideNote);
+
+    wavesurfer.on('region-play', function(region) {
+        region.once('out', function() {
+            wavesurfer.play(region.start);
+            wavesurfer.pause();
+        });
+    });
+
+    // // Zoom slider
```
Comment (thijstriemstra) [inline example/video-annotation/main.js]: remove the dead code.

File: example/media/nasa.mp4 (hunk 1 above)
Comment (xx205) [reply example/media/nasa.mp4]: This is a soft link to the mp4 file. I will replace 'nasa.mp4' with the actual file and remove the file named 'Happy Thanksgiving From the International Space Station.mp4'; the reference link will be given.

File: example/video-annotation/index.html (hunk 2 above)
Comment (xx205) [reply example/video-annotation/index.html]: OK, I will remove all the dead codes and pull again later.

File: example/media/nasa.mp4 (hunk 1 above)
Comment (xx205) [reply example/media/nasa.mp4]: Hi, I have updated the files, please review them again.

Comment (thijstriemstra) [reply example/media/nasa.mp4]: how big is the mp4 file? can you make sure it's as small as possible?

Comment (xx205) [reply example/media/nasa.mp4]: The size of nasa.mp4 has been reduced to 4.1MB.

File: example/video-annotation/main.js [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+    wavesurfer.load(mediaElt);
+
+    wavesurfer.on('ready', function() {
+        wavesurfer.enableDragSelection({
+            color: randomColor(0.25)
+        });
+
+        wavesurfer.util
+            .ajax({
```
Comment (thijstriemstra) [inline example/video-annotation/main.js]: please use `util.fetchFile` instead.

File: example/video-annotation/main.js [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+            wavesurfer.pause();
+        });
+    });
+
+    /* Toggle play/pause buttons. */
+    var playButton = document.querySelector('#play');
+    var pauseButton = document.querySelector('#pause');
+    wavesurfer.on('play', function() {
+        playButton.style.display = 'none';
+        pauseButton.style.display = '';
```
Comment (thijstriemstra) [inline example/video-annotation/main.js]: use `block` instead of `''`.

File: example/video-annotation/main.js (hunk 7 above)
Comment (xx205) [reply example/video-annotation/main.js]: Done.

File: example/video-annotation/main.js (hunk 8 above)
Comment (xx205) [reply example/video-annotation/main.js]: Done.

**Pull request merged**